src/agent.rs

```diff
@@ -300,49 +318,53 @@ impl AgentThread {
         debug!("handle request");
         let response = match request.kind {
             RequestKind::AllocMR(param) => {
-                let mr = Arc::new(
-                    self.inner.allocator.alloc(
-                        &Layout::from_size_align(param.size, param.align)
-                            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?,
-                    )?,
-                );
+                let mr = self.inner.allocator.alloc(
+                    &Layout::from_size_align(param.size, param.align)
+                        .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?,
+                )?;
                 let token = mr.token();
                 let response = AllocMRResponse { token };
                 // the MR is newly created, it's impossible to find it in the map
                 let _old = self.inner.mr_own.lock().await.insert(token, mr);
                 ResponseKind::AllocMR(response)
             }
             RequestKind::ReleaseMR(param) => {
-                assert!(self
-                    .inner
+                self.inner
                     .mr_own
                     .lock()
                     .await
                     .remove(&param.token)
-                    .is_some());
+                    .map_or_else(
+                        || debug!("{:?} already released", param.token),
```

Comment:
I guess this should be a warning? It's double release, right?

Reply:
you are right : )